aws-java-sdk-dynamodb/src/test/java/com/amazonaws/services/dynamodbv2/document/utils/ValueMapAndWithJsonSupportTest.java

```diff
@@ -0,0 +1,44 @@
+package com.amazonaws.services.dynamodbv2.document.utils;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.*;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+
+import com.amazonaws.AmazonClientException;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.Test;
+
+public class ValueMapAndWithJsonSupportTest {
+
+    private static final String NO_JSON_STRING = "nojson";
+    private static final String KEY = "somekey";
+
+    @Test(expected = AmazonClientException.class)
+    public void valueMapCreationShouldFailIfNoJsonStringIsUsedAsValue() {
+        new ValueMap().withJSON("a", NO_JSON_STRING);
+    }
+
+    @Test
+    public void valueMapShouldReturnAProperDeserializedJsonObject() {
+        Map<String, Object> expected = new LinkedHashMap<String, Object>();
+        expected.put("fruit", "pair");
+        expected.put("color", "green");
+        ObjectMapper mapper = new ObjectMapper();
+
+        try {
+            ValueMap valueMap = new ValueMap().withJSON(KEY, mapper.writeValueAsString(expected));
+            Map<String,Object> actual = (Map<String,Object>)valueMap.get(KEY);
+            assertThat(actual, is(expected));
+
+        } catch (JsonProcessingException e) {
```

Comment:
Don't bother catching this, converting to a `RuntimeException` and re-throwing, just add the `throws Exception` to the test method signature.